appengine/pond_storage/opponents.py

```diff
@@ -19,27 +19,115 @@
 
 import cgi
 import json
+import math
+import random
 from google.appengine.ext import ndb
 from pond_storage import *
 
-def get_opponent_ducks(duck):
-  """Returns list of opponent ducks."""
-  # TODO: Add filter for published ducks
-  # TODO: Add logic here to pick the duck based on the ranking
-  user_key = get_user_key(users.get_current_user())
-  duck_query = Duck.query()
-  duck_list = []
-  for duck in duck_query:
-    if duck.key.parent() != user_key:
-      duck_list.append(get_duck_info(duck))
-  return duck_list[0:3]
+START_PERCENT = .10
+END_PERCENT = .15
+NEARBY_PERCENT = .10
+DUCKS_NEEDED = 3
+
+def get_entries_in_range(user_duck, entries, start_rank, end_rank):
+  """Get all entries in the given range that"""
+  user_entry = user_duck.leaderboard_entry_key.get()
+  entryQuery = LeaderboardEntry.query()
```

Comment:
Along that line, I think it would be useful to pass the Leaderboard entity object so you can use get_entries_query and wouldn't need to pass total size (since you can just get it on the object)